src/Albany_KokkosTypes.hpp

```diff
@@ -39,13 +39,11 @@ using ViewLR = Kokkos::View<DT,Kokkos::LayoutRight,MemSpace>;
 // NOTE: Tpetra may use a different LO type (Albany uses int32, while tpetra uses int). When extracting local views/matrices,
 //       be careful about this. At worst, you may need to extract pointers and reinterpret_cast them.
 
-using DevLayout = PHX::Device::array_layout;
-
 // kokkos 1d and 2d views to be used for on-device kernels
 template<typename Scalar, typename MemoryTraits = Kokkos::MemoryUnmanaged>
-using DeviceView1d = Kokkos::View<Scalar*, DevLayout, PHX::Device, MemoryTraits>;
+using DeviceView1d = Kokkos::View<Scalar*, Kokkos::LayoutLeft, PHX::Device, MemoryTraits>;
 template<typename Scalar, typename MemoryTraits = Kokkos::MemoryUnmanaged>
-using DeviceView2d = Kokkos::View<Scalar**, DevLayout, PHX::Device, MemoryTraits>;
+using DeviceView2d = Kokkos::View<Scalar**, Kokkos::LayoutLeft, PHX::Device, MemoryTraits>;
```

Comment:
Does this layout get used on host in a serial build? I think that would affect cpu performance.

Reply:
Not sure how this would work with `getNonconstDeviceData` since that's basically just grabbing a write-accessible view. We'd have to copy the data in and then copy the data back at some point with another call.

Reply:
Ah that's weird that left layout is used always. Since the phalanx and thyra layouts are different, I think we should distinguish them here too. Is `DeviceView1d`/2d used only for thyra views? If it is, I would name this `ThyraDeviceView 1d` and grab the layout directly from Thyra.

Reply:
I had the same thought. But then realized that Tpetra uses exclusively LayoutLeft. I agree that there may be some confusion though, so something along the line of what Jerry suggested may help.

Reply:
I dug a bit more and I think these layouts should be the same. It seems like the get routines return a tpetra view which is defined by `KokkosNode`: https://github.com/sandialabs/Albany/blob/master/src/Albany_KokkosTypes.hpp#L29
which use `PHX::Device`. Maybe we can discuss the issue you're seeing offline, I don't think this should be LayoutLeft all the time.

Reply:
a lot of this seems overdefined too. shouldn't we be able to define these with just `Kokkos::View<Scalar*, PHX::Device>`?

Reply:
The Tpetra MV's use LayoutLeft to ensure that enries of the same column are contiguous, which ensures we can extract a col from an MV, and get a contiguous array.

Reply:
ah okay I remember that now. It looks like `DeviceView` is getting used for other things so we should definitely make a distinction to avoid others using it for anything other than tpetra MV.

Reply:
This is probably what we should be using in that case: https://github.com/trilinos/Trilinos/blob/master/packages/tpetra/core/src/Tpetra_MultiVector_decl.hpp#L473
via Tpetra_MultiVector::device_view_type

Reply:
I found this when looking at the Tpetra docs:

https://docs.trilinos.org/dev/packages/tpetra/doc/html/classTpetra_1_1MultiVector.html#ae9efb48f9e44a6376862b4d5e436ee87

If I understand this correctly, the underlying dual view of a Tpetra_Multivector will always have LayoutLeft.

Reply:
right, you can see it in the link I sent as well. I would make the generic `DeviceView = Kokkos::View<Scalar*, PHX::Device>` and create a new definition for the unique `TpetraMVDeviceView = Tpetra_MultiVector::device_view_type`

Reply:
or `ThyraMVDeviceView` since we still use thyra.

Reply:
> right, you can see it in the link I sent as well

Ah sorry, I should have hit refresh before I commented. I didn't see all the follow up comments